api/src/main/scala/com/lightbend/lagom/internal/api/MessageSerializers.scala

```diff
@@ -30,16 +30,30 @@ trait UnresolvedMessageSerializer[MessageEntity] extends MessageSerializer[Messa
   override def serializerForResponse(acceptedMessageHeaders: util.List[MessageProtocol]): NegotiatedSerializer[MessageEntity, Any] =
     throw new NotImplementedError("Cannot use unresolved message serializer")
 
-  def resolve(factory: SerializerFactory, typeInfo: Option[Type]): MessageSerializer[MessageEntity, _]
+  def resolve(factory: SerializerFactory, typeInfo: Type): MessageSerializer[MessageEntity, _]
+}
+
+/**
+ * Placeholder serializer to instruct Lagom to find the serializer from the method ref.
+ */
+class MethodRefMessageSerializer[MessageEntity] extends MessageSerializer[MessageEntity, Any] {
+  override def serializerForRequest(): NegotiatedSerializer[MessageEntity, Any] =
+    throw new NotImplementedError("MethodRefMessageSerializer is just a placeholder")
+
+  override def deserializer(protocol: MessageProtocol): NegotiatedDeserializer[MessageEntity, Any] =
+    throw new NotImplementedError("MethodRefMessageSerializer is just a placeholder")
+
+  override def serializerForResponse(acceptedMessageProtocols: util.List[MessageProtocol]): NegotiatedSerializer[MessageEntity, Any] =
```

Comment:
Shouldn't we use PCollections here? (I know `TreePVector` implements `List`, but I thought being stricter and requiring immutable, thread-safe, type is better).